quantus/metrics/axiomatic_metrics.py

```diff
@@ -146,27 +145,38 @@ def __call__(
             >> scores = metric(model=model, x_batch=x_batch, y_batch=y_batch, a_batch=a_batch_saliency, **{}}
         """
         # Reshape input batch to channel first order:
-        self.channel_first = kwargs.get("channel_first", get_channel_first(x_batch))
-        x_batch_s = get_channel_first_batch(x_batch, self.channel_first)
+        if "channel_first" in kwargs and isinstance(kwargs["channel_first"], bool):
+            channel_first = kwargs.pop("channel_first")
+        else:
+            channel_first = utils.infer_channel_first(x_batch)
+        x_batch_s = utils.make_channel_first(x_batch, channel_first)
 
-        # Wrap the model into an interface.
+        # Wrap the model into an interface
         if model:
-            model = get_wrapped_model(model, self.channel_first)
+            model = utils.get_wrapped_model(model, channel_first)
 
         # Update kwargs.
         self.kwargs = {
             **kwargs,
             **{k: v for k, v in self.__dict__.items() if k not in ["args", "kwargs"]},
         }
-        self.nr_channels = kwargs.get("nr_channels", np.shape(x_batch_s)[1])
-        self.img_size = kwargs.get("img_size", np.shape(x_batch_s)[-1])
+        self.perturb_baseline = self.kwargs.pop("perturb_baseline", "black")
```

Comment:
I popped twice, as most of the tests pass the same params from the `__init__` to the `__call__` again.
We could also pop  only at the end, right after updating the `self.kwargs` in the `__call__` method .

I see now, that there could be a problem with the approach I implemented here:
If one sets the perturb baseline in the `__init__`, but not in the `__call__`, it will be overwritten by the default value in `__call__`. We surely don't want this to happen.

The need for popping `perturb_baseline` comes from the explicit pass later on to `perturb_func`. If `perturb_baseline` is both passed explicitly and present in the passed `kwargs`, then an error will be raised (passed the same argument two times)

Solutions could maybe be the following:
* don't pass `perturb_baseline` explicitly to `self.perturb_func` but just via `**kwargs`
* replace `self.perturb_baseline = self.kwargs.pop("perturb_baseline", "black")` with `self.perturb_baseline = self.kwargs.pop("perturb_baseline", self.perturb_baseline)`
This would then potentially be filled with the default value setting from `__init__`. the pop wouldn't be necessary in `__init__` but also not harmful.

When this `perturb_kwargs`, `explain_kwargs`, etc. will be implemented, this will have to be adjusted again. As we then don't need to know anymore about `self.perturb_baseline` in the metrics, as it could be passed via `perturb_kwargs`.
So maybe the easiest fix for now will be removing the pops and just passing `pertub_baseline` via `**self.kwargs`.